javascript/code-challenges/challenges-08.test.js

```diff
@@ -0,0 +1,332 @@
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named containsW that takes in a string. This function should use a regular expression pattern to return true if the string contains the letter 'w' in lower case or false if it does not. 
+
+------------------------------------------------------------------------------------------------ */
+
+const containsW = (str) => {
+  let wCheck = /w/;
+  return wCheck.test(str);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named isNum that takes in a string or number of any length. This function should use a regular expression pattern to return true if the input contains a number, and false if the input does not contain a number.
+
+For example:
+12345 returns true
+'12345' returns true
+'h3llo world' returns true
+'hello world' returns false
+------------------------------------------------------------------------------------------------ */
+
+const isNum = (input) => {
+  let numCheck = /[0-9]/;
+  return numCheck.test(input);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named containsWorld that takes in a string or number of any length. This function should use a regular expression pattern to return true if the input contains the word 'world' all in lower-case letters, and false if the input does not.
+
+------------------------------------------------------------------------------------------------ */
+
+const containsWorld = (input) => {
+  let worldCheck = /world/;
+  return worldCheck.test(input);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named isCapitalized that takes in a string. This function should use a regular expression pattern to match all words that begin with a capital letter. It should only match words, not punctuation.
+
+Return an array containing all the matches.
+------------------------------------------------------------------------------------------------ */
+
+const isCapitalized = (str) => {
+  let capitalCheck = /[A-Z][a-z]*/g;
+  let capitalWasChecked = str.match(capitalCheck);
+  if (capitalWasChecked) {
+    return capitalWasChecked;
+  } else {
+    return [];
+  }
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 6
+
+Write a function named citiesAtoJ that takes in an array of city names and uses a regular expression pattern to return a new array containing any cities that begin with the letters A through J, inclusive.
+------------------------------------------------------------------------------------------------ */
+
+const citiesAtoJ = (arr) => {
+  let aThruJCities = [];
+  let aThruJCheck = /(?<! )[A-J][a-z]*/g;
```

Comment:
Hello there, nice work on challenges!

You mentioned Boston wasn't being added. Check your Regex Expression. I ran this again in the browser and changed the expression for proper output. 

- we can remove the (?<!) `lookaround`  and the `[a-z]`
- Perhaps we can use `/^[A-J]\w+/` 
- we start with `/^` which is first a delimiter and then we say we are starting at beginning of line/string
- Next we specify the range of characters uppercase (because if we look at test, all strings in the test arr are uppercase and instructions above show `A` and `J` (we want cities A-J) only
- Next we work with the `\w` to specify  word character: ASCII letter
- Last, we have `+/` delimiter